python/CMakeLists.txt

```
@@ -561,6 +572,19 @@ if(PYARROW_BUILD_CUDA)
   set_source_files_properties(pyarrow/_cuda.pyx PROPERTIES CYTHON_API TRUE)
 endif()
 
+# Acero
+if(PYARROW_BUILD_ACERO)
+  if(PYARROW_BUNDLE_ARROW_CPP)
+    bundle_arrow_lib(${ARROW_ACERO_SHARED_LIB} SO_VERSION ${ARROW_SO_VERSION})
+    if(MSVC)
+      bundle_arrow_import_lib(${ARROW_ACERO_IMPORT_LIB})
+    endif()
+  endif()
+
+  set(DATASET_LINK_LIBS ArrowAcero::arrow_acero_shared)
```

```suggestion
  set(ACERO_LINK_LIBS ArrowAcero::arrow_acero_shared)
```

And we need the following change:

```diff
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 580857d50c..e9b22d7531 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -779,9 +779,13 @@ if(PYARROW_BUILD_SUBSTRAIT)
   target_link_libraries(_substrait PRIVATE ${SUBSTRAIT_LINK_LIBS})
 endif()
 
+if(PYARROW_BUILD_ACERO)
+  target_link_libraries(_acero PRIVATE ${ACERO_LINK_LIBS})
+  target_link_libraries(_exec_plan PRIVATE ${ACERO_LINK_LIBS})
+endif()
+
 if(PYARROW_BUILD_DATASET)
   target_link_libraries(_dataset PRIVATE ${DATASET_LINK_LIBS})
-  target_link_libraries(_exec_plan PRIVATE ${DATASET_LINK_LIBS})
   if(PYARROW_BUILD_ORC)
     target_link_libraries(_dataset_orc PRIVATE ${DATASET_LINK_LIBS})
   endif()
```